### PR TITLE
[TRAFODION - 2345] Support LOG() function of any base

### PR DESCRIPTION
**Pull request by ryzuo**
Like Oracle does, the function LOG() function supports 2 parameter of call, in this way, the 1st parameter is the radix, and the second one is power, the function returns the logarithm.
This change implement this, and still, the user can call it with only one parameter (form LOG(X)), in which case the parameter X will be the power, and the function just returns the natural logarithm of X. This is also compatible with MySQL.

### CODE REVIEW

File: core/sql/exp/exp_math_func.cpp [hunk 1]
```diff
@@ -337,8 +338,22 @@ ex_expr::exp_return_type ExFunctionMath::eval(char *op_data[],
 	  **diagsArea << DgString0("LOG");
 	  return ex_expr::EXPR_ERROR;
 	}
-      
-      *(double *)op_data[0] = MathLog(*(double *)op_data[1], err);
+
+      if(3 == getNumOperands())
+    {
```
Comment (DaveBirdsall) [inline core/sql/exp/exp_math_func.cpp]: The indentation is inconsistent. Perhaps your editor is using tabs? Best to replace those with spaces.

File: core/sql/exp/exp_math_func.cpp [hunk 2]
```diff
@@ -337,8 +338,22 @@ ex_expr::exp_return_type ExFunctionMath::eval(char *op_data[],
...
+        double power = *(double *)op_data[2];
+        double radix = *(double *)op_data[1];
+        if(power <= 0 || 1 == radix)
```
Comment (DaveBirdsall) [inline core/sql/exp/exp_math_func.cpp]: The radix also has to be greater than zero, doesn't it?

File: core/sql/exp/exp_math_func.cpp (hunk 1 above)
Comment (ryzuo) [inline core/sql/exp/exp_math_func.cpp]: I use space, but somehow the original indentation is not recognized, I see they are the same in my editor, not sure which part of the configuration of vim causes this. Anyway, I tried to re-org it, will push on to see how it looks.

File: core/sql/exp/exp_math_func.cpp (hunk 2 above)
Comment (ryzuo) [inline core/sql/exp/exp_math_func.cpp]: you are right, I will change this.

**Comment by zellerh** [comment]
+1
Looks good to me (with the fix you mentioned).

**Comment by ryzuo** [comment]
The indention looks good now this time. I anything still goes wrong, please point out.

**Pull request merged**